src/incubator/WheelPicker/index.tsx

```diff
@@ -163,7 +164,13 @@ const WheelPicker = ({
     // https://github.com/facebook/react-native/issues/26661
     if (Constants.isAndroid && prevIndex.current !== index) {
       prevIndex.current = index;
-      onChange?.(items?.[index]?.value, index);
+      
+      if (prevInitialValue.current !== initialValue) {
+        // don't invoke 'onChange' if 'initialValue' changed
+        prevInitialValue.current = initialValue;
```

Comment:
I wouldn't set the `prevInitialValue` value here just because `scrollToIndex` method is being called in several places.
Instead I'd set in the `useEffect` that handle the initialValue update (right after triggering the scroll)
```

Reply:
Not sure I understan. In all other cases it should not pass the condition...

Reply:
Yes but technically the responsibility for re-setting the `prevInitialValue` is in the relevant useEffect and not in the method it calls to. 
If someone will change the implementation of `scrollToIndex` they might break the functionality of `prevInitialValue`

Reply:
I don't see how it works... Let's talk when you're available